src/main/java/com/diplomaProject/StudyBe/Subject/Service/SubjectService.java

```diff
@@ -1,10 +1,17 @@
 package com.diplomaProject.StudyBe.Subject.Service;
 
 
+import com.diplomaProject.StudyBe.Subject.Subject;
+import com.diplomaProject.StudyBe.Subject.web.dto.SubjectDto;
+import com.diplomaProject.StudyBe.User.User;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @Service
 public interface SubjectService {
+    public void addSub(SubjectDto subject);
 
+    public Subject findByName(String name);
```

Comment:
Quick not in backend when you write a function name to get data and the return is not nullable, make sure to write the function as get example getByName. Find implies that the function can return null or empty set.